airflow/config_templates/default_airflow.cfg

```diff
@@ -59,6 +59,12 @@ sql_alchemy_conn = sqlite:///{AIRFLOW_HOME}/airflow.db
 # The encoding for the databases
 sql_engine_encoding = utf-8
 
+# Collation for `dag_id`, `task_id`, `key` columns in case they have different encoding.
+# This is particularly useful in case of mysql with utf8mb4 encoding because
+# primary keys for XCom table has too big size and `sql_engine_collation_for_ids` should
+# be set to `utf8mb3_general_ci`
+# sql_engine_collation_for_ids =
```

Comment:
Great change and I do not want to delay it but maybe it would be worth to add some tests for collation?

Reply:
Well -  we already (kind of) have all the needed tests on system level. See here: https://github.com/apache/airflow/pull/7570/files#diff-bfc61c22ff7c425f1ee55fedc125fea8R24  - what happens during the tests:

-  we are using the variable to set value for that parameter
- then we run the migration scripts from alembic to create the schema from the scratch (with the collation set)
- finally all the tests are run using the DB created this way.

Before adding the collation parameter the whole thing broke down at the DB initialisation with "Index larger that 3072".

Reply:
Thanks for the answer. If tests fail already without this change this is all we need.